beacon-chain/p2p/fork.go

```diff
@@ -0,0 +1,127 @@
+package p2p
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/sirupsen/logrus"
+)
+
+// ENR key used for eth2-related fork data.
+const eth2EnrKey = "eth2"
+
+// EnrForkID represents a special value ssz-encoded into
+// the local node's ENR for discovery purposes. Peers should
+// only connect if their enrForkID matches.
+type EnrForkID struct {
```

Comment:
maybe we add this definition to our proto schema instead ? then define the appropriate ssz tags there

Reply:
It's not used anywhere else and just having [4]byte will allow us to not need ssz tags. It's a one-of struct.